Sources/TuistGenerator/Linter/TargetLinter.swift

```diff
@@ -139,4 +140,28 @@ class TargetLinter: TargetLinting {
 
         return []
     }
+
+    private func lintDeploymentTarget(target: Target) -> [LintingIssue] {
+        guard let deploymentTarget = target.deploymentTarget else {
+            return []
+        }
+
+        let issue = LintingIssue(reason: "The version of deployment target is incorrect", severity: .error)
+
+        let osVersionRegex = "\\b[0-9]+\\.[0-9]+(?:\\.[0-9]+)?\\b"
+        switch deploymentTarget {
+        case let .iOS(version, _): if !version.matches(pattern: osVersionRegex) { return [issue] }
+        case let .macOS(version): if !version.matches(pattern: osVersionRegex) { return [issue] }
+        }
+        return []
+    }
+}
+
+extension String {
+    func matches(pattern: String) -> Bool {
```

Comment:
I think this function is a good candidate to be in the `TuistCore` framework where other extensions and utilities exist. What do you think?

Reply:
Sounds good 💪